pages/workflow/PKPWorkflowHandler.inc.php

```diff
@@ -90,7 +90,7 @@ function access($args, $request) {
 		$workingStageId = null;
 
 		for ($workingStageId = $currentStageId; $workingStageId >= WORKFLOW_STAGE_ID_SUBMISSION; $workingStageId--) {
-			if (isset($accessibleWorkflowStages[$workingStageId]) && array_intersect($editorialWorkflowRoles, $accessibleWorkflowStages[$workingStageId])) {
+			if (array_key_exists($workingStageId, $accessibleWorkflowStages)) {
```

Comment:
I think this might inadvertently pass users in when `isset($accessibleWorkflowStages[$workingStageId] = array()`. The `array_intersect` will evaluate false in that case, but `array_key_exists` will evaluate true.

Reply:
Yep, I don't know why this change is still in our code. I'm going to revert it. Thanks!